sklearn/calibration.py

```diff
@@ -408,6 +447,14 @@ def fit(self, X, y, sample_weight=None):
             self.label_encoder_.fit(self.classes)
 
         self.classes_ = self.label_encoder_.classes_
+        X = np.array(X) if isinstance(X, list) else X
```

Comment:
This line is needed to make new test `test_calibration_pipeline` pass which has been added to solve issue 8710
https://github.com/scikit-learn/scikit-learn/issues/8710